ddtrace/utils/cache.py

```diff
@@ -0,0 +1,98 @@
+from threading import RLock
+
+
+_CACHED_OBJECTS = []
```

Comment:
this is only used to keep track of all cache objects so we don't need the original references in order to look them up/invalidate them?

Reply:
This is only used during test runs as I couldn't find a way to monkey patch the caching decorators before they are applied. This shouldn't normally be used to invalidate the caches as invalidations are already done where appropriate

Reply:
I looked deeper into it and found out that the pytest plugin does an early import of `ddtrace` stuff, so loading a plugin with a command line option allows intercepting and patching the `cached` decorator before it's actually applied. This way I was able to move this to test sources.